framework/filters/PageCache.php

```diff
@@ -154,57 +161,74 @@ public function beforeAction($action)
             $this->dependency = Yii::createObject($this->dependency);
         }
 
-        $properties = [];
-        foreach (['cache', 'duration', 'dependency', 'variations'] as $name) {
-            $properties[$name] = $this->$name;
-        }
-        $id = $this->varyByRoute ? $action->getUniqueId() : __CLASS__;
         $response = Yii::$app->getResponse();
-        ob_start();
-        ob_implicit_flush(false);
-        if ($this->view->beginCache($id, $properties)) {
+        $data = $this->cache->get($this->calculateCacheKey());
+        if (!is_array($data) || !isset($data['cacheVersion']) || $data['cacheVersion'] !== 1) {
+            $this->view->cacheStack[] = $this;
+            ob_start();
+            ob_implicit_flush(false);
             $response->on(Response::EVENT_AFTER_SEND, [$this, 'cacheResponse']);
             Yii::trace('Valid page content is not found in the cache.', __METHOD__);
             return true;
         } else {
-            $data = $this->cache->get($this->calculateCacheKey());
-            if (is_array($data)) {
-                $this->restoreResponse($response, $data);
-            }
-            $response->content = ob_get_clean();
+            $this->restoreResponse($response, $data);
             Yii::trace('Valid page content is found in the cache.', __METHOD__);
             return false;
         }
     }
 
     /**
-     * Restores response properties from the given data
-     * @param Response $response the response to be restored
-     * @param array $data the response property data
+     * This method is invoked right before caching a response is to be started.
+     * You may override this method to cancel caching by returning `false` or store an additional data
+     * in a cache entry by returning an array instead of `true`.
+     * @return boolean|array whether to cache or not, return an array instead of `true` to store an additional data.
+     * @since 2.0.11
+     */
+    public function beforeCacheResponse()
+    {
+        return true;
+    }
+
+    /**
+     * This method is invoked right after restoring a response is finished (but before the response is sent).
```

Comment:
Isn't it better to change spelling like this:

`This method is invoked right after the response restoring is finished`